server/text_generation_server/models/custom_modeling/flash_phi_modeling.py

```diff
@@ -0,0 +1,420 @@
+import torch
+import torch.distributed
+
+from torch import nn
+from transformers.activations import ACT2FN
+from transformers.configuration_utils import PretrainedConfig
+from typing import Optional, List, Tuple
+
+from text_generation_server.utils import paged_attention, flash_attn
+from text_generation_server.utils.layers import (
+    TensorParallelRowLinear,
+    TensorParallelColumnLinear,
+    TensorParallelEmbedding,
+    PositionRotaryEmbedding,
+    TensorParallelHead,
+    get_linear,
+    FastLayerNorm,
+)
+
+class PhiConfig(PretrainedConfig):
```

Comment:
The config is missing some values like `partial_rotary_factor`.
See: https://huggingface.co/microsoft/phi-2/blob/main/configuration_phi.py#L136